libr/bin/p/bin_dis.c

```diff
@@ -0,0 +1,302 @@
+/* radare2 - MIT - Copyright 2023 - keegan */
+
+#define R_LOG_ORIGIN "bin.dis"
+
+#include <r_types.h>
+#include <r_util.h>
+#include <r_lib.h>
+#include <r_bin.h>
+
+#include "../../arch/p/dis/dis.h"
+#include "../../arch/p/dis/dis.c"
+
+static bool check_buffer(RBinFile *bf, RBuffer *buf) {
+	ut64 pos = r_buf_tell (buf);
+	r_buf_seek (buf, 0, R_BUF_SET);
+	st32 magic;
+	if (!dis_read_operand (buf, &magic)) {
+		return false;
+	}
+	r_buf_seek (buf, pos, R_BUF_SET);
+	return magic == XMAGIC || magic == SMAGIC;
```

Comment:
maybe add some size limits to avoid false positives too. is the 4 byte magic the only valid signature here?